.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,37 @@
+name: CI Test Suite
+
+on: 
+    push:
+        branches: 
+          - main
+    pull_request: 
+        branches: 
+          - main
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: 📥 Checkout Code
+              uses: actions/checkout@v4
+
+            - name: 🏗️ Set Up Go
+              uses: actions/setup-go@v5
+              with:
+                  go-version: 1.23
+
+            - name: 📦 Install Dependencies
+              run: go mod tidy
+
+            - name: 📥 Install go-testreport
+              run: |
+                go install github.com/becheran/go-testreport@v0.2.2
```

Comment:
No need to start a multiline string, a single line should be enough for such simple commands.